server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java

```diff
@@ -169,6 +170,15 @@ private static boolean isValidField(MappedFieldType fieldType) {
         if (fieldType.indexOptions() == IndexOptions.NONE) {
             return false;
         }
+        // and must not be under nested field
```

Comment:
We should check all levels in the path, for instance in `foo.bar.baz`, `foo` can be a simple object field and `foo.bar` a nested field.